capsules/extra/src/ieee802154/phy_driver.rs

```diff
@@ -0,0 +1,460 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright Tock Contributors 2024.
+
+//! IEEE 802.15.4 userspace interface for configuration and transmit/receive.
+//!
+//! Implements a userspace interface for sending and receiving raw IEEE 802.15.4
+//! frames.
+//!
+//! Sending - Userspace fully forms the 15.4 frame and passes it to the driver.
+//!
+//! Receiving - The driver receives 15.4 frames and passes them to the process.
+//! To accomplish this, the process must first `allow` a read/write ring buffer
+//! to the kernel. The kernel will then fill this buffer with received frames
+//! and schedule an upcall upon receipt of the first packet.
+//!
+//! The ring buffer provided by the process must be of the form:
+//!
+//! ```text
+//! | read index | write index | user_frame 0 | user_frame 1 | ... | user_frame n |
+//! ```
+//!
+//! `user_frame` denotes the 15.4 frame in addition to the relevant 3 bytes of
+//! metadata (offset to data payload, length of data payload, and the MIC len).
```

Comment:
where in `user_frame` are these 3 relevant bytes?

Reply:
The answer is on line 354, but that won't be exported to documentation. Worth just repeating that comment here IMO.